examples/plot_glm_retinal_ganglia.py

```diff
@@ -0,0 +1,287 @@
+# estimator for the filter in a GLM / LNP model (as long as the nonlinearity
+# results in an STA whose expectation is not zero)
+#
+# It's useful to visualize the STA (even if your stimuli are
+# not white noise), just because if we don't see any kind of structure then
+# this may indicate that we have a problem (e.g., a mismatch between the
+# design matrix and binned spike counts.
+
+t_bins = np.arange(n_t + 1) * dt
+spikes_binned, _ = np.histogram(spike_time, t_bins)
+sta = Xdsgn.T.dot(spikes_binned) / n_spikes
+
+t_lag = dt * np.arange(-n_t_filt + 1, 1)  # time bins for STA (in seconds)
+plt.plot(t_lag, t_lag * 0, 'k--')
+plt.plot(t_lag, sta, 'bo-')
+plt.title('Spike-Triggered Average (STA)')
+plt.xlabel('Time before spike (sec)')
+plt.xlim([t_lag.min(), t_lag.max()])
+plt.show()
+
+########################################################
+# **Fitting and predicting with a linear-Gaussian GLM**
+#
+# For a general linear model, an observed spikes can be
+# thought of an underlying parameter
+# :math:`\theta` that control the spike output:
+# :math:`y = \vec{\theta} \cdot \vec{x} + \epsilon` and
+# :math:`y \sim \text{Poiss}(\vec{\theta} \cdot \vec{x})`
+#
+# We can add an offset or a "constant" to our design matrix.
+# This can be done by concatenating a column of 1 to
+# our previously created design matrix.
+
+Xdsgn_offset = np.hstack((np.ones((n_t, 1)), Xdsgn))
+
+wsta_offset = inv(Xdsgn_offset.T @ Xdsgn_offset)\
+    @ Xdsgn_offset.T.dot(spikes_binned)
+
+const, wsta_offset = wsta_offset[0], wsta_offset[1:]
+spikes_pred_lgGLM_offset = const + (Xdsgn @ wsta_offset)
+
+########################################################
+# **Fitting and predicting with a Poisson GLM**
+#
+# We can also assume that their is a non-linear function governing
+# the underlying the firing patterns. Concreately, we can write down as
+# :math:`y = f(\vec{\theta} \cdot \vec{x}) + \epsilon`, and
+# :math:`y \sim \text{Poiss}(\vec{\theta} \cdot \vec{x})`.
+# We call :math:`f^{-1}` a "link function".
+# Here, we can use Pyglmnet's `GLM` to predict the parameters.
+
+# create possion GLM instance
+glm_poisson = GLM(distr='poisson',
+                  verbose=False, alpha=0.05,
+                  max_iter=1000, learning_rate=0.2,
+                  score_metric='pseudo_R2',
+                  reg_lambda=1e-7, eta=4.0)
+
+# fitting to a design matrix
+glm_poisson.fit(Xdsgn, spikes_binned)
+
+# predict spike counts using Poisson GLM
+# alternatively, you can also use
+# np.exp(glm_poisson.beta0_ + X.dot(glm_poisson.beta_))
+spikes_pred_poissonGLM = glm_poisson.predict(Xdsgn)
+
+#############################################################################
+# **Adding spikes history for predicting spike counts**
+#
+# We can even further predict the spikes by using the spikes history.
+# **Note** the spike-history portion of the design
+# matrix had better be shifted so that we aren't allowed to use the spike
+# count on this time bin to predict itself!
```

Comment:
I think this need to be introduced earlier when you make the hankel matrix. Otherwise it's not clear why you are doing it.

Reply:
I mean the text of doing prediction from spike history. The code can stay here.

Reply:
Added some explanation on how to create a design matrix here.